core/store/src/main/java/uk/gov/gchq/gaffer/store/StoreProperties.java

```diff
@@ -461,4 +461,25 @@ public void updateStorePropertiesClass(final Class<? extends StoreProperties> re
 
         return properties;
     }
+
+    public static StoreProperties resolveStoreProperties(final Store store, final StoreProperties properties, final String parentStorePropertiesId) {
+        StoreProperties rtn = null;
+
+        if (null != parentStorePropertiesId) {
+            rtn = store.getGraphLibrary().getProperties(parentStorePropertiesId);
+        }
+        if (null != properties) {
+            if (null == rtn) {
+                rtn = properties;
+            } else {
+                // delete the old properties id as we are about to modify the properties
+                rtn.getProperties().remove(StoreProperties.ID);
+                rtn.getProperties().putAll(properties.getProperties());
+            }
+        }
+        if (null == rtn) {
+            rtn = store.getProperties();
```

Comment:
I don't think rtn = store.getProperties(); is the expected behaviour for all cases. It might be required for the federated store but probably not for AddStoreProperties.

Reply:
yes this a bug that was corrected in other branches, but not yet merged. This moving of the method would have re-introduced/moved/hid the bug. thanks